distutils/_msvccompiler.py

```diff
@@ -178,15 +178,30 @@ def _find_exe(exe, paths=None):
     return exe
 
 
-# A map keyed by get_platform() return values to values accepted by
-# 'vcvarsall.bat'. Always cross-compile from x86 to work with the
-# lighter-weight MSVC installs that do not include native 64-bit tools.
-PLAT_TO_VCVARS = {
-    'win32': 'x86',
-    'win-amd64': 'x86_amd64',
-    'win-arm32': 'x86_arm',
-    'win-arm64': 'x86_arm64',
-}
+def _get_plat_to_vcvars():
+    # A map keyed by get_platform() return values to values accepted by
+    # 'vcvarsall.bat'.
+    if get_platform() == get_host_platform() == "win-arm64":
```

Comment:
```suggestion
    if get_host_platform() == "win-arm64":
```

No reason to trust `get_platform` here, and it's used later on to look up the specific result from this dict anyway.